Can't compile

Hi, I am back.

When I try to call the following code,

```
        static INIT: Once = ONCE_INIT;
        INIT.call_once(|| unsafe {
            ffi::bcm_host_init();
            ffi::vcos_init();
            ffi::mmal_vc_init();
        });
```

I get this error,

```
  = note: /opt/vc/lib/libmmal_vc_client.so: undefined reference to `mmal_queue_destroy'
          /opt/vc/lib/libmmal_vc_client.so: undefined reference to `mmal_log_category'
          /opt/vc/lib/libmmal_vc_client.so: undefined reference to `mmal_queue_get'
          /opt/vc/lib/libmmal_vc_client.so: undefined reference to `mmal_status_to_string'
          /opt/vc/lib/libmmal_vc_client.so: undefined reference to `mmal_port_alloc'
          /opt/vc/lib/libmmal_vc_client.so: undefined reference to `mmal_buffer_header_driver_data'
          /opt/vc/lib/libmmal_vc_client.so: undefined reference to `mmal_event_format_changed_get'
          /opt/vc/lib/libmmal_vc_client.so: undefined reference to `mmal_component_action_trigger'
          /opt/vc/lib/libmmal_vc_client.so: undefined reference to `mmal_format_extradata_alloc'
          /opt/vc/lib/libmmal_vc_client.so: undefined reference to `mmal_queue_put'
          /opt/vc/lib/libmmal_vc_client.so: undefined reference to `mmal_buffer_header_replicate'
          /opt/vc/lib/libmmal_vc_client.so: undefined reference to `mmal_port_free'
          /opt/vc/lib/libmmal_vc_client.so: undefined reference to `mmal_ports_alloc'
          /opt/vc/lib/libmmal_vc_client.so: undefined reference to `mmal_component_supplier_register'
          /opt/vc/lib/libmmal_vc_client.so: undefined reference to `mmal_buffer_header_release'
          /opt/vc/lib/libmmal_vc_client.so: undefined reference to `mmal_format_copy'
          /opt/vc/lib/libmmal_vc_client.so: undefined reference to `mmal_event_error_send'
          /opt/vc/lib/libmmal_vc_client.so: undefined reference to `mmal_component_action_register'
          /opt/vc/lib/libmmal_vc_client.so: undefined reference to `mmal_port_event_get'
          /opt/vc/lib/libmmal_vc_client.so: undefined reference to `mmal_4cc_to_string'
          /opt/vc/lib/libmmal_vc_client.so: undefined reference to `mmal_port_event_send'
          /opt/vc/lib/libmmal_vc_client.so: undefined reference to `mmal_port_buffer_header_callback'
          /opt/vc/lib/libmmal_vc_client.so: undefined reference to `mmal_ports_free'
          /opt/vc/lib/libmmal_vc_client.so: undefined reference to `mmal_queue_create'
          /opt/vc/lib/libmmal_vc_client.so: undefined reference to `mmal_queue_length'
          collect2: error: ld returned 1 exit status
```

If I can still compile the code which includes the mmal-sys library and even reference other functions and constants but this initialization code fails for some reason.

--- a/build.rs
+++ b/build.rs
@@ -9,8 +9,8 @@ use std::path::PathBuf;
 
 fn main() {
     // Tell cargo to tell rustc to link the system shared libraries.
-    println!("cargo:rustc-link-lib=mmal_core");
     println!("cargo:rustc-link-lib=mmal_util");
+    println!("cargo:rustc-link-lib=mmal_core");
     println!("cargo:rustc-link-lib=mmal_vc_client");
     println!("cargo:rustc-link-lib=vcos");
     println!("cargo:rustc-link-lib=bcm_host");

